drivers/serial/uart_nrf_sw_lpuart.c

```diff
@@ -296,14 +348,19 @@ static void activate_rx(struct lpuart_data *data)
 		return;
 	}
 
+	LOG_DBG("RX: Ready");
+	data->rx_got_data = false;
+	data->rx_state = RX_ACTIVE;
 	err = uart_rx_enable(data->uart, data->rx_buf,
 				data->rx_len, data->rx_timeout);
 	__ASSERT(err == 0, "RX: Enabling failed (err:%d)", err);
 
 	/* Ready. Confirm by toggling the pin. */
-	rdy_pin_blink(data);
-	LOG_DBG("RX: Ready");
-	data->rx_state = RX_ACTIVE;
+	if (rdy_pin_blink(data) == false) {
+		/* If tranmitter behaves abnormally deactivate RX. */
+		rdy_pin_suspend(data);
```

Comment:
If you suspend the RDY pin here, how will the receiver re-activate when the transmitter is finally up and behaving correctly? Wouldn't it miss all interrupts on that line?

Reply:
suspend will disable interrupt on rdy pin, then we disable uarte. When uarte is disabled (RX_DISABLED event is received) rdy pin interrupt is reactivated (rdy pin goes to idle state). In idle state rdy pin interrupt is configured to level high detection so if transmitter wants to transmit req pin will be kept high and it will be detected.

Reply:
Makes sense. I missed the RX_DISABLED event part.

Reply:
I have a loopback test where pins (REQ and TX) are randomly toggled and it passes after those fixes.